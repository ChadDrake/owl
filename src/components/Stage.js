import React from "react";

function Stage(props) {
  if (props.onStage) {
    return (
      <div>
        <img src={props.avatar} alt="" />
        {props.name}
      </div>
    );
  }
  return null;
}

export default Stage;
