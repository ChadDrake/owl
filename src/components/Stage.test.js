import React from 'react';
import ReactDOM from 'react-dom';
import Stage from './Stage';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Stage />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Stage renders UI as expected', () => {
  const tree = renderer
    .create(
      <Stage id="h" name="Jim" avatar="img" inSession={false} onStage={true} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
