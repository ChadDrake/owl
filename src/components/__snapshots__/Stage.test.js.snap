// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage renders UI as expected 1`] = `
<div>
  <img
    alt=""
    src="img"
  />
  Jim
</div>
`;

exports[`renders UI as expected 1`] = `
<div>
  <section
    className="ParticipantsList"
  >
    <div>
      Jim
    </div>
    <img
      alt=""
      src="img"
    />
    <p>
      Onstage: 
      yes
    </p>
    <p>
      Here: 
      no
    </p>
  </section>
</div>
`;
